tekton-proposal.md

```diff
@@ -0,0 +1,73 @@
+
+## Background
+
+Knative Build was conceived at the beginning of the Knative project alongside the other serverless-focused components (Serving and Eventing) to enable a "serverless development experience". Its goal was to enable users to specify their source to a Knative Service deployment and the steps to take to make it a container, and deploy that container, while interacting with a single custom resource.
+
+From the beginning, Build had bigger ambitions. Its [2018 roadmap](https://github.com/knative/build/blob/master/roadmap-2018.md) mentions features that aren't directly related to furthering serverless deployments (triggering, workflow, etc.). These aren't accelerators for serverless deployments, these are CI/CD features!
+
+This tension has existed in Build from the beginning: did it exist to accelerate the Knative serverless platform? Or is it supposed to be a full-featured CI/CD platform? How should Build prioritize features that don't have anything to do with the rest of Knative? These are not hypothetical questions, these are questions we've heard from users, operators and contributors.
+
+To further add confusion, in August 2018 Knative began the Pipelines effort in the knative/build-pipeline repo. This was an experimental ground-up design to build on the experience with the Build resource, and build real CI/CD workflow infrastructure running on Kubernetes. Its [2019 roadmap](https://github.com/knative/build-pipeline/blob/master/roadmap-2019.md) (the knative/build repo never had a 2019 roadmap) included all the CI/CD features from 2018 and more, and makes only [passing mention](https://github.com/knative/build-pipeline/blob/master/roadmap-2019.md#dont-break-serving) of its relationship with Serving.
+
+Around the same time, Knative Serving added duck-typing support, which meant any Kubernetes resource that satisfies its barebones interface could be used in a pre-deployment build, and all of Build, TaskRun and PipelineRun satisfy that interface and can be used.
+
+Since the inception of the build-pipeline repo, most active development has happened there (with contributions from Pivotal, CloudBees, Red Hat and others). At this point, Knative Build is largely stable, requiring less than a single full-time engineer effort for requested features, high-priority bug fixes, and scheduled releases. This is most likely a good thing! With the CI/CD features moving to Pipelines, there just aren't many serverless-focused features left for Build to implement at this time, and it could largely be considered "done" until it gets new feature requests from Knative's users and operators.
+
+Pipelines contributors' ambitions continue to expand to address the market for CI/CD for deployment to platforms outside of Kubernetes: raw VMs, mobile devices, IoT devices, and the Kubernetes platform itself. Pipelines may primarily run _on Kubernetes_ but it's not necessarily _for Kubernetes_.
+
```

Comment:
> Pipelines _may_ primarily run _on Kubernetes_ 

Does this sentence mean Tekton and the pipelines could possibly run outside Kubernetes? Or rather like mentioned before it, pipelines might include tasks and actions that perform things outside Kubernetes.

Reply:
> Does this sentence mean Tekton and the pipelines could possibly run outside Kubernetes?

[A while back](https://docs.google.com/document/d/1e7gMVFlJfkFdTcaWj2qETeRD9kSBG2Vh8mASPmQMYC0/edit#heading=h.4i5oxhadeij2) SAP demo'd a prototype where the Knative Build controller was replaced with a shim to proxy the 'build' resource to another CI system (https://github.com/sap/infrabox).

Since CRDs/resources are decoupled from the controller you can swap out of the implementation.